src/LegacyRecord.php

```diff
@@ -361,34 +373,40 @@ protected static function getDataMapperInstance()
         return $instance->getDataMapper();
     }
 
-    public function getCollection(): Collection
-    {
-        return $this->getDataMapper()
-            ->getCollection();
-    }
-
     /**
-     * @return array|object
-     * @throws BindingResolutionException
+     * Handle dynamic method calls into the model.
+     *
+     * @param mixed $method     name of the method that is being called
+     * @param mixed $parameters parameters of $method
+     *
+     * @throws BadMethodCallException in case of invalid methods be called
+     *
+     * @return mixed
      */
-    public function bsonSerialize(): object|array
+    public function __call(mixed $method, mixed $parameters)
```

Comment:
```suggestion
    public function __call(mixed $method, mixed $parameters): mixed
```
Maybe here you can define mixed as return type.